### ISSUE TITLE
cmdhist only allows a single keybinding for (each of) next and prev

### ISSUE THREAD

**Issue by taobert** [luakit/luakit#913]
tl;dr: Skip to (4) below.

`cmdhist` only allows a single key to be bound to `history_prev_func`/`history_next_func`.
#641 exported the keys to bind (via `_M.history_prev`/`_M.history_next`),
so users are able to choose the (single) binding which works best for them,
however being able to have multiple bindings would be an advantage.
(For example, _i_ would like to be able to use either `<Up>`/`<Down>` 
or `<control-p>`/`<control-n>` depending on where my hand happens to be.)

To achieve multiple bindings, a few implementation strategies could be used.
The first two (and the current binding implementation)
have the downside that they are inconsistent with the "standard" (`mode.add_binds`) key bindings, 
which means that they require somewhat esoteric `cmdhist`-specific knowledge both to add the binding,
and to know what bindings there are (as they don't show up in `luakit://binds`).

1. The smallest change to `cmdhist.lua` would be to export `_M.history_prev_func`/`_M.history_next_func` via
```
--- Go to prev cmdhist item
_M.history_prev_func = history_prev_func

--- Go to next cmdhist item
_M.history_next_func = history_next_func

```
to allow users to add bindings by analogy to how it's done in cmdhist.lua:
```
window.add_signal("init", function (w)
    w:add_signal("mode-entered", function ()
        local mode = w.mode
        if mode and mode.history then
            local cmdhist = require("cmdhist")
            lousy.bind.add_bind(w.binds, "<control-p>", { func = cmdhist.history_prev_func })
            lousy.bind.add_bind(w.binds, "<control-n>", { func = cmdhist.history_next_func })
        end
    end)
end)
```

2. Slightly more impactful on `cmdhist.lua`, but less so on the userconf.lua,
    would be to expand `_M.history_prev`/`_M.history_next` processing to allow users to pass in lists of keys.
   This would require testing whether or not the passed bindings are a table to avoid breaking current config file.

3. Augmenting the exported prev/next functions to check that they've been called in a sensible context,
and using the `all` meta-mode to hold these bindings mitigates the quirks with the current implementation
and the first two points above (the bindings are defined consistently with others, and are presented in luakit://binds).
This _does_ have the downside that the bindings are only listed under "all mode",
so if, for example, a person were looking under "search mode" to find all the available key bindings, they may be missed.
I've mocked this up as a draft PR (#912) for perusal.

4. Adding a `cmdhist` meta-mode to hold bindings, 
which are then applied to other appropriate modes (those with a history table).
This has the benefits of (3), and also allows for the bindings to be displayed in the relevant sections of luakit://binds.
The (potential) downside of this is polluting the mode-space,
but to me it feels worth it for the extra functionality.
I've also mocked this up as a draft PR (#911) for perusal.

What are other peoples feelings?
